### PR TITLE
Bug: Refactor footnote to a component

### PR DESCRIPTION
**Pull request by clmedders**
## Summary

Addressing bug that was causing content to extend past content wrapper 

<details>
  <summary>Before and After</summary>
Screenshot 2024-07-25 at 10 58 52 AM
Screenshot 2024-07-25 at 10 59 13 AM
</details>

## Solution

_**TL;DR:** Updated `footer` element within the file to be a `div` to match up with best practices and applied styles to the list item to help with word wrapping_

1. Address a few other issues with markup and styles 
1a. The footnote was being applied as a footer, this is not best practice since we are already using a footer 
1b. Updated the footer tag to a div and added BEM naming to this new div and added new styles to the `li`to help with the word wrapping

This solution was chosen for a few reasons, the style solution was to help with the wrapping of the link to help address the issue with content extending past the wrapper and the update to the footer was updated to a div using BEM to align with our projects best practices and also help with mark up and remove a footer that was not truly the footer.

## How To Test

1. visit [preview page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/cm-footnote-update/resources/delivering-digital-first-public-experience/) and adjust screen size to mobile and verify that there is no longer an issue with content going past the wrapper 

Tagging supervisor - @rayestrada

### CODE REVIEW

File: content/resources/21st-century-integrated-digital-experience-act.md [hunk 1]
```diff
@@ -587,11 +587,11 @@ Contact OMB’s Office of the Federal CIO at [ofcio@omb.eop.gov](mailto:ofcio@om
 
 <br>
 
-<footer>
-<h3 id="footnote-label">Citations</h3>
+<div class="dg-footnote">
+<h3 id="footnote-label">Footnotes</h3>
```
Comment (clmedders) [inline content/resources/21st-century-integrated-digital-experience-act.md:591]: This was the only file that was not using `Footnotes` so made this update to help it be consistent with others.

File: content/guides/public-policy/_index.md [hunk 2]
```diff
@@ -75,9 +75,9 @@ Keep reading to learn more about the different [types of policies](https://digit
 
 ---
 
-<footer>
+<div class="dg-footnote">
 <h3 id="footnote-label">Footnotes</h3>
```
Comment (nick-mon1) [inline content/guides/public-policy/_index.md:79]: Is it possible to BEM-ify this `dg-footnote` into a component for the `h3`, `ol` and `li` elements?

File: themes/digital.gov/src/scss/new/styles.scss [hunk 3]
```diff
@@ -175,6 +175,7 @@
 @forward "shortcodes/card-prompt";
 @forward "shortcodes/termsofservice-compatible";
 
+@forward "footnote.scss";
```
Comment (nick-mon1) [inline themes/digital.gov/src/scss/new/styles.scss:178]: We can update to read `@forward "footnote";`

**Review by RileySeaburg — DISMISSED** [review]
Overall I think this is a great fix and addition. 

I'm not so particular about the BEM stuff but if @nick-mon1 thinks it's necessary go ahead and adjust that before we merge.

File: themes/digital.gov/src/scss/new/_footnote.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+@use "uswds-core" as *;
+
+.dg-footnote {
+  li {
+    overflow-wrap: anywhere;
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_footnote.scss:5]: **question:** Have we tested this with nested lists?

File: themes/digital.gov/src/scss/new/_footnote.scss [hunk 5]
```diff
@@ -0,0 +1,7 @@
+@use "uswds-core" as *;
+
+.dg-footnote {
+  li {
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_footnote.scss:4]: **polish:** There's only a single style, so no need to nest.

```suggestion
.dg-footnote li {
```

File: themes/digital.gov/src/scss/new/_footnote.scss (hunk 4 above)
Comment (klin2020) [reply themes/digital.gov/src/scss/new/_footnote.scss:5]: Tested with nested lists, confirming that it works with nested lists.

**Single List**
single_list

**Nested List**
nested_list

**Review by RileySeaburg — DISMISSED** [review]

**Review by nick-mon1 — DISMISSED** [review]
### Improvement

I formatted the `html` in `markdown` for lists with more than 1 item.

**Review by RileySeaburg — DISMISSED** [review]
Very thorough great job everybody!

**Review by ToniBonittoGSA — APPROVED** [review]

**Pull request merged**